sdk/eventhub/Azure.Messaging.EventHubs/src/Metadata/PartitionConsumerMetrics.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs.Metadata
+{
+    /// <summary>
+    ///   A set of information about the state of a partition, as observed by the consumer.
+    /// </summary>
+    ///
+    public class PartitionConsumerMetrics
+    {
+        /// <summary>
+        ///   The name of the Event Hub where the partitions reside, specific to the
+        ///   Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; }
+
+        /// <summary>
+        ///   The identifier of the partition, unique to the Event Hub which contains it.
+        /// </summary>
+        ///
+        public string Id { get; }
+
+        /// <summary>
+        ///   The sequence number of the last observed event to be enqueued in the partition.
+        /// </summary>
+        ///
+        public long? LastEnqueuedSequenceNumber { get; private set; }
+
+        /// <summary>
+        ///   The offset of the last observed event to be enqueued in the partition.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   The offset is the relative position for event in the context of the stream.  The offset
+        ///   should not be considered a stable value, as the same offset may refer to a different event
+        ///   as events reach the age limit for retention and are no longer visible within the stream.
+        /// </remarks>
+        ///
+        public long? LastEnqueuedOffset { get; private set; }
+
+        /// <summary>
+        ///   The date and time, in UTC, that the last observed event was enqueued in the partition.
+        /// </summary>
+        ///
+        public DateTimeOffset? LastEnqueuedTime { get; private set; }
+
+        /// <summary>
+        ///   Indicates whether or not the partition is currently empty.
+        /// </summary>
+        ///
+        /// <value>
+        ///   <c>true</c> if the partition is empty; otherwise, <c>false</c>.
+        /// </value>
+        ///
+        public DateTimeOffset? MetricsLastReceived { get; private set;  }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionConsumerMetrics"/> class.
+        /// </summary>
+        ///
+        /// <param name="name">The name of the Event Hub that contains the partitions.</param>
+        /// <param name="partitionId">The identifier of the partition.</param>
+        ///
+        protected internal PartitionConsumerMetrics(string name,
```

Comment:
I would suggest renaming the `name` to `eventHubName` so that we can save a trip to the ref docs to figure out what the name is for as otherwise it might mean (at a quick glance) that the name is for the "PartitionConsumerMetrics"

Reply:
Same for other constructors